samtranslator/translator/translator.py

```diff
@@ -137,6 +140,15 @@ def translate(  # noqa: too-many-branches
         macro_resolver = ResourceTypeResolver(sam_resources)
         intrinsics_resolver = IntrinsicsResolver(parameter_values)
 
+        # The available managed policies aren't expected to change during the
+        # run of a transform; cache function return value as it can be called
+        # multiple times
+        @lru_cache(maxsize=None)
+        def cached_get_managed_policy_map() -> Optional[Dict[str, str]]:
+            if not get_managed_policy_map:
+                return None
+            return get_managed_policy_map()
```

Comment:
I am not a fan of line functions, neither sure whether `@lru_cache` will work inside another function, as every time this is called, it creates another cached `cached_get_managed_policy_map`. I don't think caching is a concern of this part of the code, but the caller of `samtranslator` library.

Reply:
Another concern is memory leak of `lru_cache`. What if `lru_cache` still keep the cache even when the function is out of scope?

Reply:
Yeah, it's not pretty. Included as forgetting it could mean a lot of IAM calls, and weaving the caching into the instances got messy. I removed for now and will check whether I can put it in the instances more cleanly.